content/source/docs/cloud/workspaces/variables.html.md

```diff
@@ -1,130 +1,139 @@
+!> **Warning:** We recommend talking to HashiCorp support before setting `TFE_PARALLELISM`.
 
-You can also use the optional [Terraform Cloud Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/variable) to update a workspace's variables. This has the same effect as managing workspace variables manually or via the API, but can be more convenient for large numbers of complex variables.
+### Terraform Variables
 
-## Variable Limits
+Terraform variables refer to [input variables](/docs/language/values/variables.html) that define parameters without hardcoding them into the configuration. For example, you could create variables that let users specify the number and type of Amazon Web Services EC2 instances they want to provision with a Terraform module.
 
-The following limits apply to variables:
+``` hcl
+variable "instance_count" {
+  description = "Number of instances to provision."
+  type        = number
+  default     = 2
+}
+```
 
-Component   |  Limit
-------------|---------------------------
-description |  512 characters
-key         |  128 characters
-value       |  256 kilobytes
+You can then reference this variable in your configuration.
 
-## Managing Variables in the UI
+``` hcl
+module "ec2_instances" {
+  source = "./modules/aws-instance"
 
-To view and manage a workspace's variables, navigate to that workspace and click the "Variables" navigation link at the top.
+-  instance_count = 2
++  instance_count = var.instance_count
+  ## ...
+}
+```
 
-The variables page has separate lists of Terraform variables and environment variables:
+If a required input variable is missing, Terraform plans in the workspace will fail and print an explanation in the log.
 
-![Screenshot: The initial appearance of a workspace's variables page](./images/vars.png)
+#### Loading Variables from Files
 
-To edit a variable, click one of its text fields or its pencil (edit) icon to reveal the editing controls. Make any desired changes to the variable's name, value, description, and settings, then click the "Save Variable" button.
+You can provide default variable values by committing any number of [files ending in `.tfvars`](/docs/language/values/variables.html#variable-files) to workspaces that use Terraform 0.10.0 or later. If any variables from these files have the same key as existing variables in the Terraform Cloud workspace, the variables applied to the workspace overwrite variables from the files.
 
-To add a variable, click the "+ Add Variable" button, enter a name, value, optional description, and save.
+One exception is `terraform.tfvars` files. Terraform Cloud creates a `terraform.tfvars` file to pass a workspace's input variables to Terraform during runs. This means that Terraform Cloud will overwrite any `terraform.tfvars` file you check into version control. You can check in other types of `.tfvars` files, but do not check in `terraform.tfvars` files even when running Terraform solely on the command line.
```

Comment:
So I went to the expert this morning - @chrisarcand and we worked on a fix for this together! I am confident that it's correct now and also not as confusing. In short, *auto.tfvars is the only type of file you can use right now, so we're very explicit about that. And we took out the extraneous stuff about terraform.tfvars because the user doesn't really need to know it - they just need to know what does work and that they can't use any other type of file.